bin/improver-nowcast-extrapolate

```diff
@@ -0,0 +1,89 @@
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to extrapolate input data given advection velocity fields."""
+
+import datetime
+import os
+import numpy as np
+
+from improver.argparser import ArgParser
+from improver.nowcasting.optical_flow import AdvectField
+from improver.utilities.filename import generate_file_name
+from improver.utilities.load import load_cube
+from improver.utilities.save import save_netcdf
+
+
+def main():
+    """Extrapolate data forward in time."""
+
+    parser = ArgParser(
+        description="Extrapolate input data to required lead times.")
+    parser.add_argument("input_filepath", metavar="INPUT_FILEPATH",
+                        type=str, help="Path to input NetCDF file.")
+    parser.add_argument("output_dir", metavar="OUTPUT_DIR", type=str,
+                        help="Directory to write output files.")
+    parser.add_argument("--eastward_advection_filepath", type=str, help="Path"
+                        " to input file containing Eastward advection "
+                        "velocities.")
+    parser.add_argument("--northward_advection_filepath", type=str, help="Path"
+                        " to input file containing Northward advection "
+                        "velocities.")
+    parser.add_argument("--max_lead_time", type=int, default=360,
+                        help="Maximum lead time required (mins).")
+    parser.add_argument("--lead_time_interval", type=int, default=15,
+                        help="Interval between required lead times (mins).")
+    args = parser.parse_args()
+
+    # load files and initialise advection plugin
+    input_cube = load_cube(args.input_filepath)
+    ucube = load_cube(args.eastward_advection_filepath)
+    vcube = load_cube(args.northward_advection_filepath)
+
+    advection_plugin = AdvectField(ucube, vcube)
+
+    # generate list of lead times in minutes
+    lead_times = np.arange(args.lead_time_interval, args.max_lead_time+1,
+                           args.lead_time_interval)
+
+    # extrapolate input data to required lead times
+    for lead_time in lead_times:
+        # cast to float as datetime.timedelta cannot accept np.int
+        timestep = datetime.timedelta(seconds=60.*lead_time)
+        forecast_cube = advection_plugin.process(input_cube, timestep)
+
+        # save to a suitably-named output file
+        file_name = generate_file_name(forecast_cube)
```

Comment:
I know that this CLI produces multiple files and having a way of generating output names is very useful but I also think you need to be able to overwrite these output file names if required.